cpu/stm32/periph/usbdev_fs.c

```diff
@@ -179,28 +189,36 @@ static void _set_ep_out_status(uint16_t *val, uint16_t mask)
 
 static inline void _usb_attach(stm32_usbdev_fs_t *usbdev)
 {
+    const stm32_usbdev_fs_config_t *conf = usbdev->config;
+
     /* Some ST USB IP doesn't have this feature */
 #ifdef USB_BCDR_DPPU
-    const stm32_usbdev_fs_config_t *conf = usbdev->config;
     /* Enable DP pullup to signal connection */
     _global_regs(conf)->BCDR |= USB_BCDR_DPPU;
-    while (!(CRS->ISR & CRS_ISR_ESYNCF));
+    while (!(CRS->ISR & CRS_ISR_ESYNCF)) {}
 #else
-    (void)usbdev;
+    /* If configuration uses a GPIO for USB connect/disconnect */
+    if (conf->disconn != GPIO_UNDEF) {
+        gpio_set(conf->disconn);
+    }
 #endif /* USB_BCDR_DPPU */
 }
 
 static inline void _usb_detach(stm32_usbdev_fs_t *usbdev)
 {
+    const stm32_usbdev_fs_config_t *conf = usbdev->config;
+
     /* Some ST USB IP doesn't have this feature */
 #ifdef USB_BCDR_DPPU
-    const stm32_usbdev_fs_config_t *conf = usbdev->config;
     DEBUG_PUTS("usbdev_fs: Detaching from host");
 
     /* Disable DP pullup to signal disconnection */
     CLRBIT(_global_regs(conf)->BCDR, USB_BCDR_DPPU);
 #else
-    (void)usbdev;
+    /* If configuration uses a GPIO for USB connect/disconnect */
+    if (conf->disconn != GPIO_UNDEF) {
+        gpio_set(conf->disconn);
```

Comment:
`gpio_set()` is used both for attach and detach operations. Is this intentional ?

Reply:
Good catch :sunglasses: